table/tables/partition.go

```diff
@@ -156,18 +156,27 @@ func partitionRecordKey(pid int64, handle int64) kv.Key {
 // locatePartition returns the partition ID of the input record.
 func (t *PartitionedTable) locatePartition(ctx sessionctx.Context, pi *model.PartitionInfo, r []types.Datum) (int64, error) {
 	var err error
+	var isNull bool
 	partitionExprs := t.partitionExpr.UpperBounds
 	idx := sort.Search(len(partitionExprs), func(i int) bool {
 		var ret int64
-		ret, _, err = partitionExprs[i].EvalInt(ctx, chunk.MutRowFromDatums(r).ToRow())
+		ret, isNull, err = partitionExprs[i].EvalInt(ctx, chunk.MutRowFromDatums(r).ToRow())
 		if err != nil {
 			return true // Break the search.
 		}
+		if isNull {
```

Comment:
Is it a bug fix?

Reply:
Yes, a bug fix.
It's a known issue, I did not take special care to `NULL` and plan to do it later.
but I meet some problems in the test code, so I change my mind and fix it here.

Reply:
do we need to cherrypick this bugfix to release-2.0?

Reply:
I think there is no plan to support partition in release-2.0.